### PR TITLE
Add documentation for creating custom pages with Smarty.

### LINKED ISSUE #264: Add documentation for making custom Smarty template files
#### Page with Issue
https://docs.suitecrm.com/developer/views/

#### What is wrong, less clear, or missing

The page doesn't have anything mentioning the ability to create custom HTML pages using Smarty `.tpl` files. I've actually been just using PHP heredocs and putting the HTML inside my controllers thus far, and assumed it wasn't possible to separate the two. ([see this StackOverflow answer for an example of how to use the Smarty templates in a custom view](https://stackoverflow.com/a/38918990))

#### Proposed fixes and/or additions

Add a section to the Views documentation (or a whole new page) about creating custom `tpl` files, using them in view PHP files, where they should go in the `custom/` directory, etc. Separation of concerns is nice, so I think documenting this would make people's custom modules much better :)

### PR DESCRIPTION
**Pull request by connorshea**
Resolves #264. I've documented this with a simple example based on some of our own custom code and the code in https://github.com/salesagility/SuiteCRM/commit/ce06cf4ce00289b3f81dfbc72a82630aa6cca130

### CODE REVIEW

**Comment by connorshea** [comment]
This should probably be part of the existing Views documentation, I'll move it.

File: content/developer/Views.adoc [hunk 1]
```diff
@@ -163,4 +165,89 @@ most common ones to override are:
   alter this behaviour or to output anything after the main display. You
   usually want to call parent::display(); to ensure that the display
   code is run (unless, of course, you are adding your own display
-  logic). link:../views[↩]
\ No newline at end of file
+  logic). link:../views[↩]
```
Comment (pgorod) [inline content/developer/Views.adoc]: Please remove that weird character from the end of the sentence. I know you didn't put it there, it's a left-over artifact from the old wiki we had, but let's take this chance to clean it up. Thanks

File: content/developer/Views.adoc [hunk 2]
```diff
@@ -163,4 +165,89 @@ most common ones to override are:
...
+        $smarty->assign('products', $store->products);
+        $storePage = $smarty->fetch('custom/modules/Store/templates/store.tpl');
+        return $storePage;
+    }
+}
+----------
+
+And then we'll create a file at `custom/modules/Store/templates/store.tpl`, that looks like this:
+
+[source,smarty]
```
Comment (pgorod) [inline content/developer/Views.adoc]: I wonder if Asciidoc recognizes `smarty` here... it doesn't seem like it. Maybe `hmtl` produces better visual results?

File: content/developer/Views.adoc [hunk 3]
```diff
@@ -163,4 +165,89 @@ most common ones to override are:
...
+<ul>
+    {foreach name=productsIteration from=$products key=id item=product}
+        <li>{$product->name}: {$product->price}</li>
+    {/foreach}
+</ul>
+----------
+
+Much simpler!
+
+You can read more about Smarty template files and their syntax https://www.smarty.net/docsv2/en/[in the Smarty documentation].
```
Comment (pgorod) [inline content/developer/Views.adoc]: For external links please add a caret at the end, to make the link open in a new window/tab.

Like this `[in the Smarty documentation^]`

File: content/developer/Views.adoc [hunk 4]
```diff
@@ -163,4 +165,89 @@ most common ones to override are:
   alter this behaviour or to output anything after the main display. You
   usually want to call parent::display(); to ensure that the display
   code is run (unless, of course, you are adding your own display
-  logic). link:../views[↩]
\ No newline at end of file
+  logic). link:../views[↩]
+
+
+=== Customising with Smarty templates.
```
Comment (pgorod) [inline content/developer/Views.adoc]: Please remove the full stop from the end of the title

**Comment by connorshea** [comment]
@pgorod I've resolved all the problems you pointed out 👍

**Pull request merged**